Explorer/Assets/DCL/InWorldCamera/InWorldCamera/UI/InWorldCameraController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Arch.Core;
+using Cysharp.Threading.Tasks;
+using DCL.Audio;
+using DCL.CharacterCamera;
+using DCL.ExplorePanel;
+using DCL.InWorldCamera.CameraReelStorageService;
+using DCL.InWorldCamera.CameraReelStorageService.Schemas;
+using DCL.InWorldCamera.InWorldCamera.Playground;
+using DCL.UI;
+using ECS.Abstract;
+using JetBrains.Annotations;
+using MVC;
+using System.Diagnostics;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.InWorldCamera.ScreencaptureCamera.UI
+{
+    public class InWorldCameraController : ControllerBase<InWorldCameraView>
+    {
+        private readonly Button sidebarButton;
+        private readonly World world;
+        private readonly IMVCManager mvcManager;
+        private readonly ICameraReelStorageService storageService;
+
+        private ScreencaptureShortcutsController shortcutsController;
+        private SingleInstanceEntity? cameraInternal;
+
+        private bool shortcutPanelIsOpen;
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Overlay;
+        private SingleInstanceEntity? camera => cameraInternal ??= world.CacheCamera();
+
+        public InWorldCameraController([NotNull] ViewFactoryMethod viewFactory, Button sidebarButton, World world, IMVCManager mvcManager, ICameraReelStorageService storageService) : base(viewFactory)
+        {
+            this.world = world;
+            this.mvcManager = mvcManager;
+            this.storageService = storageService;
+            this.sidebarButton = sidebarButton;
+
+            sidebarButton.onClick.AddListener(RequestEnableInWorldCamera);
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            viewInstance!.CloseButton.onClick.AddListener(RequestDisableInWorldCamera);
```

Comment:
Isn't this coupling the view and the controller? Shouldn't the view provide events and hide whether they are raised by a button or any other UI element?